lib/bloc/podcast_history/podcast_history_bloc.dart

```diff
@@ -0,0 +1,184 @@
+import 'package:breez/bloc/async_actions_handler.dart';
+import 'package:breez/bloc/podcast_history/actions.dart';
+import 'package:breez/bloc/podcast_history/model.dart';
+import 'package:breez/bloc/podcast_history/sqflite/podcast_history_database.dart';
+import 'package:breez/bloc/podcast_history/sqflite/podcast_history_model.dart';
+import 'package:rxdart/rxdart.dart';
+import '../../services/injector.dart';
+
+class PodcastHistoryBloc with AsyncActionsHandler {
+  final BehaviorSubject<PodcastHistoryTimeRange> _podcastHistoryRange =
+      BehaviorSubject();
+
+  Stream<PodcastHistoryTimeRange> get posReportRange =>
+      _podcastHistoryRange.stream;
+
+  final BehaviorSubject<bool> _showShareButton = BehaviorSubject();
+
+  Stream<bool> get showShareButton => _showShareButton.stream;
+
+  final BehaviorSubject<PodcastHistoryRecord>
+      _podcastHistoryRecordBehaviourSubject = BehaviorSubject();
+
+  Stream<PodcastHistoryRecord> get podcastHistoryRecord =>
+      _podcastHistoryRecordBehaviourSubject.stream;
+
+  final BehaviorSubject<List<PodcastHistoryModel>>
```

Comment:
Can we add the podcastHistoryList to the podcastHistoryRecord? The code shows that they are calculating and changing at the same time (when the report is generated).

Reply:
Done